src/test/java/io/appium/java_client/android/AndroidFunctionTest.java

```diff
@@ -127,17 +129,23 @@ public void complexWaitingTestWithPreCondition() {
         assertThat("There should be 3 calls", calls.size(), is(3));
     }
 
-    @Test(expected = TimeoutException.class)
+    @Test
     public void nullPointerExceptionSafetyTestWithPrecondition() {
-        Wait<Pattern> wait = new FluentWait<>(Pattern.compile("Fake_context"))
-                .withTimeout(ofSeconds(30)).pollingEvery(ofMillis(500));
-        assertTrue(wait.until(searchingFunction.compose(contextFunction)).size() > 0);
+        assertThrows(TimeoutException.class,
+                () -> {
+                    Wait<Pattern> wait = new FluentWait<>(Pattern.compile("Fake_context"))
+                            .withTimeout(ofSeconds(30)).pollingEvery(ofMillis(500));
+                    assertTrue(wait.until(searchingFunction.compose(contextFunction)).size() > 0);
+                });
     }
 
-    @Test(expected = TimeoutException.class)
+    @Test
     public void nullPointerExceptionSafetyTestWithPostConditions() {
-        Wait<Pattern> wait = new FluentWait<>(Pattern.compile("Fake_context"))
-                .withTimeout(ofSeconds(30)).pollingEvery(ofMillis(500));
-        assertTrue(wait.until(contextFunction.andThen(searchingFunction).andThen(filteringFunction)).size() > 0);
+        assertThrows(TimeoutException.class,
+                () -> {
+                    Wait<Pattern> wait = new FluentWait<>(Pattern.compile("Fake_context"))
+                            .withTimeout(ofSeconds(30)).pollingEvery(ofMillis(500));
```

Comment:
it can be moved outside `assertThrows` lambda

Reply:
Updated above comments